### PR TITLE
Update to 2.5.0.1 & Pre build dev dependencies

### PR DESCRIPTION
**Pull request by aschmois**
Changes:
- Uses stack 2.5.0.1
- Prebuilds our dependencies
- Currently tries to be smart with how they're cached, might collapse into one command if the size turns out to be way bigger.

### CODE REVIEW

**Review by gera-cameron — APPROVED** [review]
So would this mean any time we add a new dep to smurf, we would want to add it here instead?

**Comment by aschmois** [comment]
> So would this mean any time we add a new dep to smurf, we would want to add it here instead?

Correct, I'm still trying out some size optimizations/investigations, I want to keep the download for developers to a minimum but I'm trying out size vs cacheability strategies.

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,20 @@
-FROM amazonlinux:2018.03.0.20180827
+FROM amazonlinux:2.0.20200722.0
 
-ARG STACK_VERSION=2.3.1
-ENV \
-  PATH=/root/.local/bin:$PATH \
-  STACK_ROOT=/stack-root
+# GHC version should match the LTS version, otherwise the path will not have the correct ghc version
+ARG GHC_VERSION=8.8.4
+ARG LTS_VERSION=lts-16.15
```
Comment (tfausak) [inline Dockerfile:5]: https://www.stackage.org/lts-16.16 :stuck_out_tongue_winking_eye:

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
 
-RUN \
-  set -o xtrace && \
-  yum update -y && \
-  yum install -y \
+ARG ITPROTV_ROOT=/home/itprotv
+ARG STACK_ROOT=$ITPROTV_ROOT/.stack
+
+ENV LANG=C.UTF-8 \
+  LC_ALL=C.UTF-8 \
```
Comment (tfausak) [inline Dockerfile:12]: Why not `en_US.UTF-8`?

**Review by tfausak — COMMENTED** [review]
There's a lot going on here. Gonna need a bit to grok it.

File: Dockerfile (hunk 2 above)
Comment (aschmois) [reply Dockerfile:12]: Copied it from: https://github.com/commercialhaskell/stackage/blob/master/automated/dockerfiles/lts-16.12/Dockerfile there has to be some reason for it.

**Comment by aschmois** [comment]
Moved to https://github.com/EdutainmentLIVE/docker-stack-develop

**Pull request closed without merging**